cpp/autosar/src/rules/A14-5-2/NonTemplateMemberDefinedInTemplate.ql

```diff
@@ -167,7 +167,9 @@ where
         mf = c.getAMemberFunction() and not mf.isCompilerGenerated() and not exists(mf.getBlock())
       )
     )
-  )
+  ) and
+  // Omit using alias (cf. https://github.com/github/codeql-coding-standards/issues/739)
+  not d instanceof UsingAliasTypedefType
```

Comment:
I think we can actually do better here. In the case we want to exclude, the entire "base type" of the using alias is considered to be template parameter. So we can exclude it directly:
```suggestion
  // Exclude Using alias which refer directly to a TypeParameter
  not d.(UsingAliasTypedefType).getBaseType() instanceof TemplateParameter
```
That removes the false positive and retains the true positive.